engine/gamesys/src/gamesys/resources/res_sound_data.cpp

```diff
@@ -225,7 +157,8 @@ namespace dmGameSystem
         uint32_t request_size   = g_SoundDataContext->m_ChunkSize;
         uint32_t request_offset = uint32_t(offset / request_size) * request_size;
 
-        SoundDataChunk* chunk = FindChunk(resource, offset);
+        ResourceCacheChunk tmp = {0};
+        ResourceCacheChunk* chunk = FindChunk(resource, offset, &tmp);
```

Comment:
The actual "reading" part is still handled by the sound resource.
Might be able to create an abstraction later that works for other filetypes as well.